qt/python/mantidqt/widgets/algorithmselector/model.py

```diff
@@ -81,3 +82,13 @@ def get_algorithm_data(self):
         unique_alg_names = set(descr.name
                                for descr in algm_factory.getDescriptors(True))
         return sorted(unique_alg_names), data
+
+    def find_input_workspace_property(self, algorithm):
+        algm_manager = AlgorithmManager.Instance()
+        alg_instance = algm_manager.createUnmanaged(algorithm[0],algorithm[1])
+        alg_instance.initialize()
+        for prop in alg_instance.getProperties():
+            if isinstance(prop, IWorkspaceProperty):
+                if prop.direction in [Direction.Input, Direction.InOut]:
```

Comment:
These could be written as a single if-statement:
`if isinstance(prop, IWorkspaceProperty) and prop.direction in [Direction.Input, Direction.InOut]:`